src/cpp/session/modules/SessionPackages.R

```diff
@@ -1226,79 +772,692 @@ if (identical(as.character(Sys.info()["sysname"]), "Darwin") &&
+   if (!is.null(pkgDesc[["biocViews"]]))
    {
-      .Call("rs_addFirstRunDoc", scratchPath, "R/hello.R", PACKAGE = "(embedding)")
+      mirror <- getOption("BioC_mirror", default = "https://bioconductor.org")
+      fmt <- "%s/packages/release/bioc/html/%s.html"
+      url <- sprintf(fmt, mirror, pkgDesc[["Package"]])
+      return(url)
    }
+   
+   # Give up.
+   ""
+})
 
-   ## NOTE: This must come last to ensure the other package
-   ## infrastructure bits have been generated; otherwise
-   ## compileAttributes can fail
-   if (usingRcpp &&
-       .rs.isPackageVersionInstalled("Rcpp", "0.10.1") &&
-       require(Rcpp, quietly = TRUE))
-   {
-      Rcpp::compileAttributes(packageDirectory)
-      if (!is.null(scratchPath) &&
-          file.exists(file.path(packageDirectory, "src/rcpp_hello.cpp")))
+.rs.addFunction("listInstalledPackages", function()
+{
+   # Look for packages in the library paths.
+   pkgPaths <- list.files(.rs.uniqueLibraryPaths(), full.names = TRUE)
+   
+   # Include only packages which have a 'Meta' sub-directory.
+   hasMeta <- file.exists(file.path(pkgPaths, "Meta"))
+   pkgPaths <- pkgPaths[hasMeta]
+   
+   # Iterate over these packages, and read their DESCRIPTION files.
+   pkgDescs <- lapply(pkgPaths, function(pkgPath) {
+      
+      pkgDesc <- tryCatch(
+         .rs.readPackageDescription(pkgPath),
+         condition = function(e) list(Package = basename(pkgPath))
+      )
+      
+      # Pull out package name for later use.
+      pkgName <- pkgDesc[["Package"]]
+      
+      # Also record metadata about the library where it was found.
+      libraryPath <- dirname(pkgPath)
+      pkgDesc[["Library"]] <- .rs.createAliasedPath(libraryPath)
+      pkgDesc[["LibraryAbsolute"]] <- libraryPath
+      pkgDesc[["LibraryIndex"]] <- match(libraryPath, .libPaths(), nomatch = 0L)
+      
+      # Also note which packages appear to be loaded or attached.
+      isLoaded <- FALSE
+      if (pkgName %in% loadedNamespaces())
       {
-         .Call("rs_addFirstRunDoc", scratchPath, "src/rcpp_hello.cpp", PACKAGE = "(embedding)")
+         isLoaded <-
+            identical(pkgDesc[["Priority"]], "base") ||
+            identical(getNamespaceInfo(pkgName, "path"), pkgPath)
       }
-   }
+      
+      isAttached <-
+         isLoaded &&
+         paste("package", pkgName, sep = ":") %in% search()
+      
+      pkgDesc[["Loaded"]] <- isLoaded
+      pkgDesc[["Attached"]] <- isAttached
+      
+      pkgDesc[["Source"]] <- tryCatch(
+         .rs.inferPackageSource(pkgDesc),
+         error = function(cnd) "[Unknown]"
+      )
+      
+      pkgDesc[["BrowseUrl"]] <- tryCatch(
+         .rs.inferPackageBrowseUrl(pkgDesc),
+         error = function(cnd) ""
+      )
+      
+      url <- pkgDesc[["URL"]]
+      if (!is.null(url))
+      {
+         urls <- strsplit(url, "[,\\s\\n]+", perl = TRUE)[[1L]]
+         pkgDesc[["PackageUrl"]] <- urls[[1L]]
+      }
```

Comment:
This differs from BrowseUrl in that it refers to a public documentation URL, as opposed to a landing page for the package's actual source.